mindsdb/interfaces/chatbot/memory.py

```diff
@@ -108,18 +108,36 @@ def _get_chat_history(self, chat_id, table_name):
         time_col = t_params['time_col']
         chat_id_cols = t_params['chat_id_col'] if isinstance(t_params['chat_id_col'], list) else [t_params['chat_id_col']]
 
-        ast_query = Select(
-            targets=[Identifier(text_col),
-                     Identifier(username_col),
-                     Identifier(time_col)],
-            from_table=Identifier(t_params['name']),
-            where=[BinaryOperation(
+        chat_id = chat_id if isinstance(chat_id, tuple) else (chat_id,)
+        # Add a WHERE clause for each chat_id column.
+        where_conditions = [
```

Comment:
as I know where have to be ASTNode object, not array:
where=BinaryOperation(op='and',
      args=[
          BinaryOperation(op='=', args=[Identifier(text_col), Constant(1)]),
          BinaryOperation(op='=', args=[Identifier(text_col), Constant(1)]),
      ]
  )

Reply:
@ea-rus You are right; I've updated it to be an instance of `BinaryOperation`. Strangely though, it looks like both forms (list or `BinaryOperation`) work here.

Reply:
Also BinaryOperation has to have only two arguments. If more arguments required it has to be nested BinaryOperations
```
          BinaryOperation(
                op='and',
                args=where_conditions  <---  len(args) == 2 
            ),
```

Reply:
Hey @ea-rus,
I've pushed another commit. Please let me know if this is correct. I've tested it and it seems to work fine.